pkg/ocm/v1/addons.go

```diff
@@ -95,28 +84,113 @@ type AddOnRequirement struct {
 }
 
 //+kubebuilder:object:generate=true
-// using list so we can easily DeepCopy
-type AddOnRequirementList struct {
-	Items []AddOnRequirement `json:"items"`
+type AddOnSubOperator struct {
+	OperatorName      string `json:"operator_name" validate:"required"`
+	OperatorNamespace string `json:"operator_namespace" validate:"required"`
+	Enabled           bool   `json:"enabled" validate:"required"`
 }
 
-func (a *AddOnRequirementList) UnmarshalJSON(b []byte) error {
-	return json.Unmarshal(b, &a.Items)
+// +kubebuilder:validation:Pattern=`^([A-Za-z -]+ <[0-9A-Za-z_.-]+@redhat\.com>,?)+$`
+type Notification string
+
+//+kubebuilder:object:generate=true
+type Monitoring struct {
+	// +kubebuilder:validation:Required
+	Namespace string `json:"namespace"`
```

Comment:
As we plan to use `go-validator` for field validation (will do a PR real soon), can we pre-emptively add `validate: "required"` tags to all required fields? Thank you. :)

Reply:
Sure thing!